OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -58,6 +58,29 @@ public static void ProcessOrder(OrderManager orderManager, World world, int clie
 						break;
 					}
 
+				case "PlaceBeacon":
+					{
+						var client = orderManager.LobbyInfo.ClientWithIndex(clientId);
+						if (client != null)
+						{
+							var player = world.FindPlayerByClient(client);
+							if (player == null)
+							{
+								if (orderManager.LocalClient.IsObserver)
+								{
+									Game.AddChatLine(Color.White, ServerChatName, "{0} placed a beacon".F(client.Name));
+									player = world.Players.Single((p) => p.PlayerName == "Everyone");
```

Comment:
Using the `Everyone` actor for this isn't very nice.  It would be much cleaner to rework `PlaceBeacon` to work with the World actor instead.

Reply:
Please drop into the IRC channel if you'd like some help, i'm happy to offer some lower level suggestions on how to do this